kb(textarea): Indicate that MaxLength is reached (#1633)
telerik/blazor-docs: knowledge-base/textarea-indicate-when-maxlength-is-reached.md

---
title: Indicate that MaxLength is reached in TextArea
description: How to indicate that MaxLength is reached in TextArea? How to display a message showing that the user has typed the maximum allowed characters? How to show a character counter?
type: how-to
page_title: Indicate that MaxLength is reached in TextArea
slug: textarea-kb-indicate-when-maxlength-is-reached
position: 

> tags: textarea, maxlength, max, count, counter, characters, message, indication

Reviewer: The tags must carry some meaning outside context

```suggestion
tags: telerik, blazor, textarea, maxlength
```

> ticketid:

Reviewer: Is this KB based on a ticket?

res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>TextArea for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

The TextArea has a `MaxLength` parameter but it does not show when the maximum allowed characters count is reached. How to notify the user they have reached the `MaxLength` value?

How to display a counter, so the user knows how many characters they have left before reaching the `MaxLength` value?

## Solution

> The `MaxLength` parameter of the TextArea maps to the `maxlength` attribute of the HTML `<textarea>` element. Its purpose is to prevent the user from typing if they reached the specified maximum allowed characters count. It does not provide visual indication out of the box.

Reviewer 2: ```suggestion
The `MaxLength` parameter of the TextArea maps to the `maxlength` attribute of the HTML `<textarea>` element. Its purpose is to prevent the user from typing beyond the specified maximum character count. It does not provide a visual indication out of the box.
```

To notify the user that they have reached the `MaxLength` value:

* Declare the desired message in an `if` block. 
* Display this message if the length of the TextArea value equals the specified `MaxLength` value. 
* Add the desired CSS rules or classes to style the message as needed.

> To let the user know how many characters they have left to type, render the TextArea value length and the `MaxLength` value - this will serve as a character counter.

Reviewer 2: ```suggestion
To inform the user about the remaining characters, render the TextArea value length and the `MaxLength` value—this will serve as a character counter.
```

>caption Show a character counter and a message when max is reached

````CSHTML
@using System.ComponentModel.DataAnnotations

@if (string.IsNullOrEmpty(FormSubmitMessage))
{
    <TelerikForm Model="@FormModel"

>                  OnValidSubmit="@OnValidSubmit"

Reviewer: I can submit the form successfully even if I exceed the `MaxLength` value.

In general, this KB should only show how to implement a counter. There is no need to mimic validation errors with custom HTML, because our customers can achieve this with built-in DataAnnotations:

```
    public class ValidationModel
    {
        [Required(ErrorMessage = "Please enter a text.")]
        [MaxLength(40, ErrorMessage = "up to 40 please")]
        public string Text { get; set; }
    }
```

                        @if (TextLength == MaxLength)
                        {
                            <div class="k-form-hint k-form-error">You reached the maximum allowed characters.</div>
                        }
                        <span class="character-counter k-form-hint @(TextLength == MaxLength ? "k-form-error" : "")">@($"{TextLength} / {MaxLength}")</span>
                    </div>
                </Template>
            </FormItem>
        </FormItems>
        <FormButtons>
            <TelerikButton ButtonType="@ButtonType.Submit" ThemeColor="primary">Send</TelerikButton>
        </FormButtons>
    </TelerikForm>
}
else
{
    @FormSubmitMessage
}

<style>
    .textarea-info {
        display: flex;
        justify-content: space-between;
    }

    .character-counter {
        margin-left: auto;
    }
</style>

@code {
    public class ValidationModel
    {
        [Required(ErrorMessage = "Please enter a text.")]
        public string Text { get; set; }
    }

    public ValidationModel FormModel = new ValidationModel();

    public string FormSubmitMessage { get; set; }

    public int TextLength => FormModel?.Text?.Length ?? 0;

>     public int MaxLength { get; set; } = 100;

Reviewer: Use a smaller `MaxLength` value, so that developers can test this more easily. Currently I have to type something, then copy-paste it multiple times.
